libs/designer-ui/src/lib/utils/utils.ts

```diff
@@ -353,3 +353,7 @@ export const getConnectorCategoryString = (connectorId: string): string => {
 
   return isBuiltInConnector(connectorId) ? builtInText : isCustomConnector(connectorId) ? customText : azureText;
 };
+
+export const convertUIElementNameToAutomationId = (uiElementName: string): string => {
+  return uiElementName === null ? '' : uiElementName.replace(/\W/g, '_').toLowerCase();
```

Comment:
Null is not common in Javascript, it is itself a valid value in Javascript, Please use this expression -
uiElementName?.replace(...)?.toLowerCase()

Also whej there is no element name please dont set the data automation ID to empty string just let it return undefined.

Reply:
Fixed this